### PR TITLE
Include elyra package in all data science workbenches

### PR DESCRIPTION
**Pull request by harshad16**
Include elyra package in all data science workbenches

## Description

This PR includes elyra package as whole to data science workbench images.
It would bring in support for elyra pipeline, extensions and plugins.

Followed the similar pattern from contrib: https://github.com/opendatahub-io-contrib/workbench-images/tree/main/jupyter

Related-to: https://github.com/opendatahub-io/notebooks/issues/55
https://github.com/opendatahub-io/s2i-lab-elyra/issues/61

## How Has This Been Tested?

To be tested from PR images.

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by atheo89** [comment]
Awesome! I have just one observation to make. Would it be possible to hide the `/utils/setup-elyra.sh` from the `/opt/app-root/src` workdir? Perhaps you could adopt the approach used in the minimal [dockerfile](https://github.com/harshad16/odh-notebooks/blob/setup-elyra/jupyter/minimal/ubi8-python-3.8/Dockerfile#L14-L16).

Screenshot from 2023-03-24 12-41-20

**Comment by harshad16** [comment]
Thanks @atheo89 , i will fix that :+1:

**Comment by harshad16** [comment]
Ready for Review:
Include elyra plugins:
pipeline-editor, python-editor and code-snippet 
reference: https://github.com/opendatahub-io/notebooks/issues/55#issuecomment-1486838468

The plugin are working in the image:
Screenshot from 2023-03-28 09-24-48

File: jupyter/datascience/ubi8-python-3.8/Dockerfile
Comment (atheo89) [inline jupyter/datascience/ubi8-python-3.8/Dockerfile:23]: It seems that the Python version displayed on the launcher is not inherited from minimal. 
Could you please add the following command on the data-science dockerfile? 

```
# Replace Notebook's launcher, "(ipykernel)" with Python's version 3.x
RUN sed -i -e "s/Python.*/$(python --version | awk '{print "Python", $2}' | cut -d'.' -f-2)\",/" /opt/app-root/share/jupyter/kernels/python3/kernel.json
```
Moreover, update the corresponding command into the minimal dockerfile, and remove it from the TrustyAi dockerfile, because it will be inherited by the datascience dockerfile. 

This cmd displays the python version without the patch version, so eventually, we could close also this issue: https://github.com/opendatahub-io/notebooks/issues/49

**Comment by atheo89** [comment]
Moreover, while I was browsing into the files, I observed that the pyTorch 3.9 notebook, has wrong references on the Lebels cmd: https://github.com/harshad16/odh-notebooks/blob/setup-elyra/jupyter/pytorch/ubi9-python-3.9/Dockerfile#L4-L12 Could you update the those to 3.8 to 3.9? 
Additionally, to that, I realized the `io.openshift.build.image` label in all the dockefiles points to the old registry, but should points to `quay.io/opendatahub/workbench-images:${NOTEBOOK_IMAGE}` :satisfied: . 
I know that these updates don't relate to that particular pr, but if it is easy for you could you update them, otherwise we could open another PR later for it.

File: jupyter/datascience/ubi8-python-3.8/Dockerfile
Comment (harshad16) [reply jupyter/datascience/ubi8-python-3.8/Dockerfile:27]: Seems like this cmd is needed explicitly in each image build.
As the jupyter setup is getting overridden in build-time, the base image change of the kernel version is not carried over.
Tested on different images. 
Included it in another images as well.

File: jupyter/datascience/ubi8-python-3.8/Dockerfile
Comment (harshad16) [reply jupyter/datascience/ubi8-python-3.8/Dockerfile]: ack, i have made the change.
:+1:

File: jupyter/datascience/ubi8-python-3.8/Dockerfile
Comment (atheo89) [reply jupyter/datascience/ubi8-python-3.8/Dockerfile]: Sure, let's leave it for later on, to unblock this pr

**Comment by VaishnaviHire** [comment]
/lgtm 

Tested this by running an [example](https://github.com/elyra-ai/examples/tree/main/pipelines/introduction-to-generic-pipelines) pipeline

Screenshot 2023-04-07 at 11 00 14 AM

**Review by VaishnaviHire — APPROVED** [review]

**Comment by shalberd** [comment]
Elyra has been taken over by RedHat, and RedHat in turn is a part of IBM, so Redhat will now also take part of discussions at elyra-ai.slack.com?

Regarding this PR: That is very cool! @harshad16 So there is no more need for an explicit Elyra notebook image via overlay? https://github.com/opendatahub-io/odh-manifests/tree/master/notebook-images/overlays/additional

**Comment by harshad16** [comment]
> Elyra has been taken over by RedHat, and RedHat in turn is a part of IBM, so Redhat will now also take part of discussions at elyra-ai.slack.com?

we will take part in the slack.

> Regarding this PR: That is very cool! @harshad16 So there is no more need for an explicit Elyra notebook image via overlay? https://github.com/opendatahub-io/odh-manifests/tree/master/notebook-images/overlays/additional

yes , the eventual goal is to remove the overlay
our plan is to wait for one release and then once it announce and user have time to switch to new images
we would remove the overlay.

**Pull request merged**
